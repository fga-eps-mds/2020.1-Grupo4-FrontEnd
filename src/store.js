import { configureStore } from '@reduxjs/toolkit';
import usersReducer from './slices/usersSlice';
import tutorialSlice from './slices/tutorialSlice'

export default configureStore({
  reducer: {
    users: usersReducer,
    tutorial: tutorialSlice
  }
});