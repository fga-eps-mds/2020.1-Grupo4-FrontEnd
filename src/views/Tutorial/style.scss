.tutorial {
  min-height: 100vh;
  max-width: 80%;
  margin: 0 auto;
  padding: 5rem 1rem;

  @media(max-width: 600px) {
    max-width: 90%;
    padding: 5rem 0;
  }

  &__header {
    display: flex;
    justify-content: space-between;

    @media(max-width: 600px) {
      padding: 0 1rem;
      flex-direction: column;
    }

    &--progress {
      @media(max-width: 600px) {
        margin: 1.25rem auto 0;
      }

      span {
        font-weight: 700;
        color: var(--secondary-green);
        text-transform: uppercase;
        padding: 0.4rem;
        border: 2px solid var(--secondary-green);
        border-radius: 10px;
      }
    }
  }

  &__body {
    display: flex;
    min-height: 50vh;
    padding: 3em;
    background: #FFFFFF;
    box-shadow: 0px 5px 10px rgba(43, 43, 43, 0.1), 0px 15px 40px rgba(0, 0, 0, 0.05);
    border-radius: 10px;
    margin: 1rem 0;

    @media(max-width: 600px) {
      margin-top: 2rem;
      flex-direction: column;
      padding: 1rem;
    }
  }
}