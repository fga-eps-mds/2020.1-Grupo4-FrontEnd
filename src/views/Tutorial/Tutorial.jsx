import React from 'react';
import './style.scss'
import Markdown from './components/Markdown/Markdown';

function Tutorial() {
  return (
  <div className="tutorial">
    <div className="tutorial__content">
      <div className="tutorial__content--header">
        <div>
          <h1>Tutorial</h1>
          <p>0% total conclu√≠do</p>
        </div>
        <div className="tutorial__content--header--progress">
          0 atividades completas
        </div>
      </div>
      <div className="tutorial__content--body">
        <Markdown />
      </div>
    </div>
  </div>
  );
}

export default Tutorial;