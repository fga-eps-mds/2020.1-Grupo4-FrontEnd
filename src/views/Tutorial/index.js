import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Switch, useRouteMatch, useLocation } from 'react-router-dom';
import { selectCurrentUser } from '../../slices/usersSlice';
import {
  selectCompletedActivities,
  selectTotalProgress,
} from '../../slices/tutorialSlice';
import { getProgress } from '../../services/tutorialServices';
import Markdown from './components/Markdown';
import ActivitiesList from './components/ActivitiesList';
import TutorialActivity from './components/TutorialActivity';
import './style.scss';

import MotionDiv from '../../UI/animation/MotionDiv';
import RouteTransition from '../../UI/animation/RouteTransition.jsx'
import { AnimatePresence, AnimateSharedLayout } from 'framer-motion'

function Tutorial({
  completedActivities,
  getProgress,
  totalProgress,
  currentUser,
}) {
  // eslint-disable-line no-shadow
  const match = useRouteMatch();
  const location = useLocation();

  useEffect(() => {
    getProgress();
  }, [getProgress]);

  return (

    <div className="tutorial">
      <div className="tutorial__header">
        <div>
          <h1>Tutorial</h1>
          {currentUser.userType === 'Learner' ? (
            <p>Total conclu√≠do: {totalProgress || 0}%</p>
          ) : (
            <p>Tutorial destinado aos aprendizes</p>
          )}
        </div>
        {currentUser.userType === 'Learner' && (
          <div className="tutorial__header--progress">
            <span>{completedActivities} atividades completas</span>
          </div>
        )}
      </div>
      <MotionDiv layout className="tutorial__body">
        {currentUser.userType === 'Learner' && <ActivitiesList />}
        <div className="custom-animation">
          <AnimatePresence exitBeforeEnter inital={false}>
            <Switch location={location} key={location.pathname} >
              <RouteTransition exact path={match.url}>
                <Markdown />
              </RouteTransition>
              <RouteTransition
                path={`${match.url}/atividades/:activityNumber`}
              ><TutorialActivity /></RouteTransition>
            </Switch>
          </AnimatePresence>
        </div>
      </MotionDiv>

    </div>
  );
}

Tutorial.propTypes = {
  completedActivities: PropTypes.number.isRequired,
  totalProgress: PropTypes.number.isRequired,
  getProgress: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  completedActivities: selectCompletedActivities(state),
  totalProgress: selectTotalProgress(state),
  currentUser: selectCurrentUser(state),
});

const mapDispatchToProps = (dispatch) => ({
  getProgress: () => dispatch(getProgress()),
});

export default connect(mapStateToProps, mapDispatchToProps, null, {
  pure: true,
})(Tutorial);
