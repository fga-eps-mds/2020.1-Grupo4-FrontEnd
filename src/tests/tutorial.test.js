import { shallow, render } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from "react-redux";
import Tutorial from '../views/Tutorial';
import { store } from "../store";

it('Expect to render Tutorial component', () => {
  expect(
    shallow(
      <MemoryRouter>
        <Tutorial />
      </MemoryRouter>
    ).length).toEqual(1)
})