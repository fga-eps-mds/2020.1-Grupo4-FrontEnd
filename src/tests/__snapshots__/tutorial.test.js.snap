// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Tutorial component 1`] = `
<div
  className="tutorial"
>
  <div
    className="tutorial__content"
  >
    <div
      className="tutorial__content--header"
    >
      <div>
        <h1>
          Tutorial
        </h1>
        <p>
          0% total conclu√≠do
        </p>
      </div>
      <div
        className="tutorial__content--header--progress"
      >
        0 atividades completas
      </div>
    </div>
    <div
      className="tutorial__content--body"
    >
      <Activities />
      <Connect(Markdown) />
    </div>
  </div>
</div>
`;
